app/models/miq_server.rb

```diff
@@ -51,18 +51,15 @@ def self.active_miq_servers
   end
 
   def self.atStartup
-    configuration  = VMDB::Config.new("vmdb")
-    starting_roles = configuration.config.fetch_path(:server, :role)
+    starting_roles = ::Settings.server.role
 
     # Change the database role to database_operations
-    roles = configuration.config.fetch_path(:server, :role)
+    roles = starting_roles.dup
     if roles.gsub!(/\bdatabase\b/, 'database_operations')
-      configuration.config.store_path(:server, :role, roles)
+      ::Settings.save!(MiqServer.my_server, :server => {:role => roles})
+      ::Settings.reload!
```

Comment:
I was not using MiqServer.setConfig since I am not sure at what point guid is created to make ```is_local?``` true to make this executed: ```  def reload_settings
    Vmdb::Settings.reload! if is_local?
  end```

Other than above, It looks like ```MiqServer.setConfig``` and ::Settings.save! + reload! will do the same since server not started and ```enqueue_for_server('reload_settings') if started?``` in MiqServer.setConfig will not be executed